lib/inputstreamhelper/__init__.py

```diff
@@ -274,13 +274,21 @@ def _update_widevine(self):
                 return
 
         wv_config = load_widevine_config()
-        latest_version = latest_widevine_version()
-        if 'x86' in arch():
+        if not wv_config:
+            log(3, 'Widevine config missing. Could not determine current version, forcing update.')
```

Comment:
I don't understand the error, if the Widevine config is missing, there is nothing to update to.
It should report and return.

Reply:
load_widevine_config reads the config of the currently installed version.

Reply:
Ah, ok. So this is the case where no Widevine was ever installed through InputStreamHelper?
But how about the user added its own Widevine CDM manually?

Looking at the code now, this is all rather confusing. We get the current_version from "select_best_chromeos_image" ?
Either we should add some comments to indicate why or the function calls need to reflect this better. Not just here, but in various other places as well.

(This is no criticism to anyone, it's just that I haven't looked at this code for a while and I got totally confused now :-))

Reply:
If a user manually places the libwidevinecdm.so in the cdm directory, he should have enough expertise to simply disable inputstreamhelper.

For ARM, the whole recovery.conf (with all the devices) is saved in the same place as the libwidevinecdm.so extracted from one of the images in that conf. The current version is determined by running select_best_chromeos_image on said recovery.conf.
I agree, that keeping the whole recovery.conf is unnecessary, but please open a new issue for that, as it's unrelated to this.